### PR TITLE
Grid display half hour courses

### PR DESCRIPTION
**Pull request by foreverwander**
Allows the grid to display classes that START on the half hour

First modified the PHL245 text file to show the half hour with added "1"

Then changed convertTimes in util to recognize the situation by adding "H" at the end of the section times (the "E" is for ending at the half hour - next step)

Also changed generate grid so the table cell IDs no longer have ".5" but "H" (for some reason ".5" doesn't work too well with rendering things...)

Finally added a function to expand the grid where an half hour course is found, and the half hour is conveniently displayed because of the "H" in its ID.

Note: currently expanding is a non-reversed process

### CODE REVIEW

**Comment by david-yz-liu** [comment]
Haven't tried it out yet, but there's some style issues in your changes that you should fix. Things like spaces in the wrong places, using `==` instead of `===`.

File: js/grid/mouse_events.js [hunk 1]
```diff
@@ -118,13 +118,48 @@ function renderClearHover(time) {
 function renderAddHover(time, section) {
     'use strict';
 
+    var n;
+
+    n = time.indexOf('H');
+
+    if(n != -1) {
```
Comment (Ian-Stewart-Binks) [inline js/grid/mouse_events.js]: Add a space between `if` and `(`.

File: js/grid/mouse_events.js [hunk 2]
```diff
@@ -118,13 +118,48 @@ function renderClearHover(time) {
...
+    } 
+
     if ($(time).attr('clicked') !== 'true') {
         $(time).html(section.courseName)
                .attr('hover', 'good');
     } else if ($(time).html() === section.courseName &&
-               $(time).attr('type') === section.type) {
-        $(time).attr('hover', 'remove');
+        $(time).attr('type') === section.type) {
+    $(time).attr('hover', 'remove');
```
Comment (Ian-Stewart-Binks) [inline js/grid/mouse_events.js]: Indentation is a bit off.

File: js/grid/mouse_events.js [hunk 3]
```diff
@@ -118,13 +118,48 @@ function renderClearHover(time) {
...
+    var cellID;
+
+    for (var k = 0; k < 5; k++) {
+        cellID = '#' + weekPrefixArray[k] + time + 'F';
+        pcells[pcells.length] = cellID;
+        cellID = '#' + weekPrefixArray[k] + time + 'S';
+        pcells[pcells.length] = cellID;
+    }
+
+    for (var k = 0; k < 5; k++) {
```
Comment (Ian-Stewart-Binks) [inline js/grid/mouse_events.js]: My JS is a bit rusty, but I don't think you need to redeclare k here.

File: js/grid/generate_grid.js [hunk 4]
```diff
@@ -142,31 +142,22 @@ function appendTableData(trFall, trSpring, time) {
...
 
-        trFall.append($('<td></td>')
-            .attr('hidden', 'true')
-            .addClass('timetable-time')
-            .html(adjustedTime));
 
         for (var k = 0; k < 5; k++) {
             trFall.append($('<td></td>')
-                .attr('id', weekPrefixArray[k] + time + 'F')
+                .attr('id', weekPrefixArray[k] + (time-0.5) + 'H' + 'F')
```
Comment (Ian-Stewart-Binks) [inline js/grid/generate_grid.js]: Add a space between `time` and `-0.5`.

File: js/common/utilities/util.js [hunk 5]
```diff
@@ -117,14 +117,25 @@ function convertTimes(times) {
...
     var time;
+    var stime;
 
     for (var i = 0; i < times.length; i++) {
         var timeString = 'MTWRF'.charAt(times[i][0]);
         time = times[i][1];
-        timeString = timeString + time;
+        if (time > 21) {
+            stime = time.toString();
+            if (stime.charAt(stime.length-1) == '0') {
```
Comment (Ian-Stewart-Binks) [inline js/common/utilities/util.js]: Add a space between `-`.

File: js/grid/mouse_events.js [hunk 6]
```diff
@@ -118,13 +118,48 @@ function renderClearHover(time) {
...
-               $(time).attr('type') === section.type) {
-        $(time).attr('hover', 'remove');
+        $(time).attr('type') === section.type) {
+    $(time).attr('hover', 'remove');
     } else {
         $(time).attr('hover', 'conflict');
     }
 }
+
+function extendRow(time) {
```
Comment (Ian-Stewart-Binks) [inline js/grid/mouse_events.js]: Please add JS doc.

Comment (Ian-Stewart-Binks) [inline js/grid/mouse_events.js]: Please add 'use strict'; statement.

**Comment by david-yz-liu** [comment]
@foreverwander You can remake the PR for the branch "half-hour"

**Pull request closed without merging**